lzero/policy/random_policy.py

```diff
@@ -165,26 +206,47 @@ def _forward_collect(
 
             for i, env_id in enumerate(ready_env_id):
                 distributions, value = roots_visit_count_distributions[i], roots_values[i]
+
+                if self._cfg.type in ['sampled_efficientzero']:
+                    try:
+                        root_sampled_actions = np.array([action.value for action in roots_sampled_actions[i]])
+                    except Exception:
+                        # logging.warning('ctree_sampled_efficientzero roots.get_sampled_actions() return list')
+                        root_sampled_actions = np.array([action for action in roots_sampled_actions[i]])
+
                 # NOTE: Only legal actions possess visit counts, so the ``action_index_in_legal_action_set`` represents
                 # the index within the legal action set, rather than the index in the entire action set.
                 action_index_in_legal_action_set, visit_count_distribution_entropy = select_action(
                     distributions, temperature=self._collect_mcts_temperature, deterministic=False
                 )
 
                 # ****** sample a random action from the legal action set ********
-                # all items except action are formally obtained from MCTS 
-                random_action = int(np.random.choice(legal_actions[env_id], 1))
+                if self._cfg.type in ['sampled_efficientzero']:
+                    random_action = self.action_space.sample()
+                else:
+                    # all items except action are formally obtained from MCTS
+                    random_action = int(np.random.choice(legal_actions[env_id], 1))
                 # ****************************************************************
-
                 # NOTE: The action is randomly selected from the legal action set, the distribution is the real visit count distribution from the MCTS seraech.
-                output[env_id] = {
-                    'action': random_action,
-                    'distributions': distributions,
-                    'visit_count_distribution_entropy': visit_count_distribution_entropy,
-                    'value': value,
-                    'pred_value': pred_values[i],
-                    'policy_logits': policy_logits[i],
-                }
+                if self._cfg.type in ['sampled_efficientzero']:
```

Comment:
you should add extra key-values in if-branch rather than add two if-else branches, which is redundant